async_service/types.py

```diff
@@ -98,6 +108,10 @@ async def publish_output_message(
     ):
         ...
 
+    @abc.abstractmethod
+    async def get_output_message(self, request_id: str) -> Optional[OutputMessage]:
```

Comment:
Should implementations be allowed to return None?

```suggestion
    async def get_output_message(self, request_id: str) -> OutputMessage:
```

Reply:
My bad, seems like this should be 
```suggestion
    async def get_output_message(self, request_id: str) -> bytes:
```
now